cmd/fs-v1-multipart.go

```diff
@@ -247,10 +331,29 @@ func (fs fsObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 		return "", err
 	}
 
-	uploadIDPath := path.Join(bucket, object, uploadID)
+	bucketDir, err := fs.getBucketDir(bucket)
+	if err != nil {
+		return "", toObjectErr(err, bucket)
+	}
+
+	if _, err = fsStatDir(bucketDir); err != nil {
+		return "", toObjectErr(traceError(err), bucket)
+	}
+
+	uploadIDPath := pathJoin(bucket, object, uploadID)
 
 	preUploadIDLock := globalNSMutex.NewNSLock(minioMetaMultipartBucket, uploadIDPath)
 	preUploadIDLock.RLock()
+	rwlk, err := fs.rwPool.Open(pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket, object, uploadsJSONFile))
+	if err != nil {
+		preUploadIDLock.RUnlock()
+		if err == errFileNotFound {
+			return "", traceError(InvalidUploadID{UploadID: uploadID})
+		}
+		return "", toObjectErr(traceError(err), bucket, object)
+	}
+	defer rwlk.Close()
```

Comment:
It is on purpose disallowed upload of parallel parts, want it to block and complete one if them completed. If needed parts can be uploaded to same server.  This optimization we can do later.